src/main/java/org/opensearch/plugin/insights/core/listener/QueryInsightsListener.java

```diff
@@ -94,18 +94,20 @@ public QueryInsightsListener(final ClusterService clusterService, final QueryIns
      * @param enabled    boolean
      */
     public void setEnableTopQueries(final MetricType metricType, final boolean enabled) {
-        boolean isAllMetricsDisabled = !queryInsightsService.isEnabled();
-        this.queryInsightsService.enableCollection(metricType, enabled);
```

Comment:
We still need to call `this.queryInsightsService.enableCollection` to enable collection for a metric, it cannot be removed.

Reply:
Got remove by mistake, good catch